### ISSUE TITLE
soft warp hides Chinese punctuation at ↪

### ISSUE THREAD

**Issue by AndyJado** [helix-editor/helix#6585]
### Summary

Chinese punctuation got hidden when it should squeezed to next line after  ↪。

tested with  ！。，#@%&

### Reproduction Steps

 

```toml
[editor.soft-wrap]
enable = true
```

I tried this:

1. `hx temp.txt`
2. i
3. 。
4. esc
5. gh
6. i
7. insert anything before 。and keep pushing it.

I expected this to happen:
。get pushed into next line
Instead, this happened:
。disappears

### Platform

mac

### Terminal Emulator

alacritty

### Helix Version

23.03-1-g406c5c38

**Comment by pascalkuthe** [comment]
Yeah this is case of wrapping one grapheme too late for double-width graphemes. There is code that handles this but that could probably use more tests I think I might have already a fix for this locally somewhere. It will probably tie in with the changes that #6417 builds on
